components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/publishers/WSO2APIPublisher.java

```diff
@@ -1,697 +1,597 @@
-                                              store.getDisplayName());
-                if (added) {   //If API creation success,then try publishing the API
-                    published = publishAPIToStore(api.getId(), store.getEndpoint(), store.getUsername(),
-                                                  httpContext,store.getDisplayName());
-                }
-                logoutFromExternalStore(store, httpContext);
-            }
-        }
-        return published;
     }
 
-    @Override
-    public boolean deleteFromStore(APIIdentifier apiId, APIStore store) throws APIManagementException {
-        boolean deleted = false;
-        if (store.getEndpoint() == null || store.getUsername() == null || store.getPassword() == null) {
-            String msg = "External APIStore endpoint URL or credentials are not defined. " +
-                         "Cannot proceed with deleting API from the APIStore - " + store.getDisplayName();
-            throw new APIManagementException(msg);
-        } else {
-            CookieStore cookieStore = new BasicCookieStore();
-            HttpContext httpContext = new BasicHttpContext();
-            httpContext.setAttribute(ClientContext.COOKIE_STORE, cookieStore);
-            boolean authenticated = authenticateAPIM(store,httpContext);
-            if (authenticated) {
-                deleted = deleteWSO2Store(apiId, store.getUsername(), store.getEndpoint(),
-                                          httpContext,store.getDisplayName());
-                logoutFromExternalStore(store, httpContext);
-            }
-            return deleted;
+    /**
+     * Exports API as an advertised API using APIImportExportManager.
+     *
+     * @param api API artifact to import
+     * @return API archive
+     * @throws APIManagementException If an error occurs while exporting API.
+     */
+    private File exportAPIArchive(API api) throws APIManagementException {
+
+        File file;
+        String tenantDomain = null;
+        int tenantId;
+        APIImportExportManager apiImportExportManager;
+
+        try {
+            apiImportExportManager = new APIImportExportManager(getLoggedInUserProvider(), getLoggedInUsername());
+            //Set additional API properties to identify it as an advertised only API
+            api.setAdvertiseOnly(true);
+            //Change owner to original provider as the provider will be overriding after importing
+            api.setApiOwner(api.getId().getProviderName());
+            //set redirect URL for the original Store
+            tenantDomain = MultitenantUtils.getTenantDomain(APIUtil.replaceEmailDomainBack(
+                    api.getId().getProviderName()));
+            tenantId = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
+                    .getTenantId(tenantDomain);
+            api.setRedirectURL(getExternalStoreRedirectURL(tenantId));
```

Comment:
Do we rewrite these data during export when passing the modified API object to 'apiImportExportManager.exportAPIArchive'?

Reply:
No, we are not rewriting. These data are written to api.json file as meta-data by parsing the API model object using Gson.

Reply:
Ref: https://github.com/wso2/carbon-apimgt/blob/3c71325ea6ac5102e5257a2d17a35016ca6d6bb7/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIExportUtil.java#L534

Reply:
Got it. Thanks